[docs] Gentler “Getting Started” with less CMS emphasis

mattstein opened the pull request:
## The Issue

Randy [suggested in Discord](https://discord.com/channels/664580571770388500/1062097685143879801/1062097685143879801) that people get discouraged when their CMS type isn’t explicitly supported, and I suggested that we adjust the “Getting Started” framing to put less emphasis on CMS options and hopefully create a bit more clarity.

## How This PR Solves The Issue

This adds a “Starting a Project” page as the first step under “Getting Started”, now leading into (and followed in the nav by) “CMS Quickstarts”. It also pushes “Performance” out of that first slot, because why did that ever make sense?

I tried to strip everything out of those first steps that didn’t _have_ to be there, and follow with support and jumping-off points below. The mention of generic `php` projects got a callout specifically because that’s the thing we need to do a better job of explaining right up front.

It also removes the “Generic” tab from the “CMS Quickstarts” page and shortens the now-redundant intro—both helping to focus that page on the CMS options it already promises to cover in its title.

## Manual Testing Instructions

[Preview changes locally](https://ddev.readthedocs.io/en/stable/developers/testing-docs/#preview-changes) or inspect the [automatic build](https://ddev--4558.org.readthedocs.build/en/4558/users/project/).

## Automated Testing Overview

n/a

## Related Issue Link(s)

https://discord.com/channels/664580571770388500/1062097685143879801/1062097685143879801

## Release/Deployment Notes

n/a


<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4558"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

tyler36 commented:
Looks good to me. On a related note:

1. I think the frameworks should be alphabetically ordered, except with "Generic" at the top. 
2. Perhaps we need a rethink of the current layout since it "hides" many supported frameworks.

![2023-01-18_11-19](https://user-images.githubusercontent.com/7234392/213066695-464d914c-6b03-4b6b-a5b4-c0d3d1e30153.png)

Maybe we should have a callout to the sidebar: "See the sidebar menu for a guide for your favorite framework."?

rfay commented:
Maybe "Generic" should be "Any" ?

mattstein commented:
> Perhaps we need a rethink of the current layout since it "hides" many supported frameworks.

I completely agree with this—it’s a problem with tabs in several places.

I also _removed_ “Generic” from this page because it’s not a CMS Quickstart:

![Screen Shot 2023-01-17 at 06 41 36 PM@2x](https://user-images.githubusercontent.com/2488775/213068939-40f7b4e6-b726-40d2-8547-fd4437c40021.png)

Ordering for the sake of the reader would be ideal, either:

1. Alphabetically as you suggested (easy to understand+navigate quickly).
2. By popularity, either platform usage or (more ideally) usage in DDEV projects if we have that information.

mattstein commented:
We could improve the issue without changing the layout if it’s reasonable to combine “OpenMage/Magento 1” and “Magento 2” under a single “Magento” tab. Wouldn’t solve it, but would help a bit.

rfay commented:
OpenMage and Magento can be combined as "Magento". Magento2 first I think.

tyler36 commented:
> Ordering for the sake of the reader would be ideal ... :

I will vote alphabetically, because its easier to understand. Reader's eyes will gravitate towards _their_ framework.

Do we have states on frameworks used? Would be interested in seeing.

mattstein commented:
I’m good with alphabetical.

FWIW I compared platforms using BuiltWith a while back [over here](https://github.com/drud/ddev/pull/4286#discussion_r993694863), for a broad idea of usage.

sreichel commented:
> OpenMage and Magento can be combined as "Magento"

Since Magento1 is EOL, i would prefer "OpenMage".

rfay commented:
I was saying that "Magento1/OpenMage" and "Magento 2" can be combined into one tab named "Magento", with separate sections explaining, with Magento 2 first.

sreichel commented:
Ah okay. Thought it was only about the tab title. Thanks.

tyler36 commented:
> OpenMage and Magento can be combined as "Magento"

Using that logic, should Backdrop be merged with Drupal?

mattstein commented:
Alphabetized what’s there without combining anything just yet:

![Screen Shot 2023-01-17 at 07 28 42 PM@2x](https://user-images.githubusercontent.com/2488775/213076735-70c273ce-51d0-4bf5-a9cc-a846cd2d8354.png)

rfay commented:
I totally think Backdrop should be merged with Drupal. Perhaps Backdrop support should be dropped too. Good call @tyler36

tyler36 commented:
>  Perhaps Backdrop support should be dropped too.

Keeping it under Drupal will help show we still support it, without taking up too much space. Drupal 7 is due to go EOL in 2023-11. If it does (has been extended twice), there might be an uptick in Backdrop users.

mattstein commented:
Just consolidated these tabs, which hopefully reflects discussion here:

- Moved “Backdrop” to level-3 heading within “Drupal” tab.
- Renamed “Magento 2” tab “Magento”, and moved “OpenMage/Magento 1” into it (removing its standalone tab).
- Removed the “6” from “Shopware 6” tab name only, leaving the “Shopware 6” heading within that tab. This removes all version numbers from tabs so they can purely describe the relevant platform irrespective of its version number(s).

![Screen Shot 2023-01-18 at 09 43 21 AM](https://user-images.githubusercontent.com/2488775/213256240-7abb4c3d-52d1-40d4-915d-29ed256d9046.gif)

rfay reviewed (changes requested):
I like this quite a lot, significant improvement.

* I have a suspicion that people will skip the new "project" page, and it may need to be named better. 
* Could the quickstarts page at least link to it in that short paragraph?
* Backdrop seemed to get orphaned somewhere? Shouldn't it be a subtab in Drupal?

mattstein commented:
> I have a suspicion that people will skip the new "project" page, and it may need to be named better.

It should at least _help_ that it’s the next page after “DDEV Installation”. It seemed like “Starting a Project” was the most straightforward, CMS-agnostic way to describe spinning up a first DDEV project—what would make that name better? (Or rather, what’s the name not doing a good job of right now?)

> Could the quickstarts page at least link to it in that short paragraph?

Good idea, will link back to it on “ready to go”.

> Backdrop seemed to get orphaned somewhere? Shouldn't it be a subtab in Drupal?

It’s a subheading rather than a tab because it seems to be a module, but I’m thinking now that I found and linked to the wrong “Backdrop” 😅:

![Screen Shot 2023-01-18 at 03 14 57 PM@2x](https://user-images.githubusercontent.com/2488775/213316071-c277f4a9-78b9-4bd3-9cda-66cf5ae6129b.png)

rfay commented:
Backdrop is a fork of Drupal 7, very much Drupal, definitely not a module.

rfay commented:
backdropcms.org

mattstein commented:
Well that’s embarrassing. Thanks @rfay, [fixed](6c276f796c150d1abb6d2e322ad47f172cc3f463)!

<img width="550" alt="Screen Shot 2023-01-18 at 03 22 39 PM@2x" src="https://user-images.githubusercontent.com/2488775/213317086-f047d8b4-7ef8-4319-ae47-688087522e25.png">

rfay commented:
Aren't you supposed to know everything about everything all the time? :)

mattstein commented:
> Aren't you supposed to know everything about everything all the time? :)

I do.\*

\*: Except for most of the things most of the time.

rfay reviewed (approved):
Looking great. Would you want Subtabs in Magento? Magento 1 aka OpenMage is as different from Magento 2 as Craft from Drupal.

rfay commented:
Or I guess I should say as far from Drupal 7 to Drupal 8, which is a long, long way.

mattstein commented:
> Would you want Subtabs in Magento? Magento 1 aka OpenMage is as different from Magento 2 as Craft from Drupal.

I don’t think so; for me those sub-tabs are an invitation to get lost more than a helpful separating mechanism.

tyler36 commented:
I really like this. 

Great work Matt!
